### PR TITLE
Move pre-commit GitHub action to pre-commit.ci

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

~Using pre-commit official GitHub action.~
Move pre-commit GitHub action to pre-commit.ci.

### Details

- pre-commit project has [official GitHub Actions of run pre-commit](https://github.com/pre-commit/action).

### CODE REVIEW

**Review by akaszynski — APPROVED** [review]

**Comment by akaszynski** [comment]
The [action website](https://github.com/pre-commit/action) states:

> generally you want to use [pre-commit.ci](https://pre-commit.ci/)which is faster and has more features.

I think we decided to not go with that because we didn't want to have it modify the files independently. I'm actually fine if we decide to have the CI action modify our source. I've seen this implemented in other repos and have yet to see issues arise.

**Comment by banesullivan** [comment]
> actually fine if we decide to have the CI action modify our source

I'd like to push back against that. If implemented, it needs to be fully opt-in.

**Comment by MatthewFlamm** [comment]
I'm more concerned about this part

> this action is in maintenance-only mode and will not be accepting new features.

This means that any change that requires significant rework will probably be the end of it.

> The [action website](https://github.com/pre-commit/action) states:

>> generally you want to use [pre-commit.ci](https://pre-commit.ci/)which is faster and has more features.

> I think we decided to not go with that because we didn't want to have it modify the files independently.

I am fairly certain that you can turn off this behavior in pre-commit.ci.  It has configuration options [autofix_prs](https://pre-commit.ci/#configuration-autofix_prs) which would turn off auto-updates to PRs.  Auto-updates to main branch opens a PR, it doesn't directly make changes, nor should it be possible since that branch is protected.  See pull requests from https://github.com/python-telegram-bot/python-telegram-bot/pulls here which do not have any autoupdate changes in PRs, but have the pre-commit.ci checks run.

**Comment by akaszynski** [comment]
Thanks for the hint @MatthewFlamm!

Since `pre-commit` recommends for you to use their integration, I modified this PR to match [python-telegram-bot/python-telegram-bot](https://github.com/python-telegram-bot/python-telegram-bot) in:
https://github.com/python-telegram-bot/python-telegram-bot/blob/53093ebceb7375716840f16a794ebd028a6effbb/.pre-commit-config.yaml#L3-L5

I've disabled `autofix_prs` and made it 100% opt-in to match @banesullivan's request. Please note that you can run this manually on a PR by commenting with:

`pre-commit.ci autofix`

**Review by MatthewFlamm — DISMISSED** [review]
I assume an administrator to this repo will have to install the app after merging.

**Comment by MatthewFlamm** [comment]
Oh wait it is working already 🥳   

I assume we can get rid of this workflow too?  https://github.com/pyvista/pyvista/blob/main/.github/workflows/pre-commit-update.yml

 Or should we wait until it sends a PR first?

**Comment by tkoyama010** [comment]
@akaszynski Thanks for finding the configuration! I had been using pre-commit-bot as the default for my repositories, but I didn't realize it could be configured in such detail.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# Integration with GitHub Actions
+# See https://pre-commit.ci/
+ci:
+    autofix_prs: false
+    autoupdate_schedule: monthly
```
Comment (tkoyama010) [inline .pre-commit-config.yaml:5]: I guess it prefers to do weekly `monthly` seems a bit leisurely :)
```suggestion
    autoupdate_schedule: weekly
```

**Comment by tkoyama010** [comment]
> I assume we can get rid of this workflow too?

@MatthewFlamm  Good catch. Thanks!

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

**Comment by akaszynski** [comment]
> pre-commit.ci autofix

Yea, I'm going to use this a lot.

**Comment by tkoyama010** [comment]
OK! All my wishes have been granted and I thank the PyVista community for their help. And welcome to our community [pre-commit-ci[bot]](https://pre-commit.com/) !

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -485,6 +485,10 @@ The actual installation of the environment happens before the first commit
 following ``pre-commit install``. This will take a bit longer, but subsequent
 commits will only trigger the actual style checks.
 
+Even if you are not in a situation where you are not performing or able to
+perform the above tasks, you can comment "pre-commit.ci autofix" on a pull
```
Comment (akaszynski) [inline CONTRIBUTING.rst:489]:
```suggestion
perform the above tasks, you can comment `pre-commit.ci autofix` on a pull
```

**Review by akaszynski — APPROVED** [review]

**Comment by banesullivan** [comment]
Okay this is quite nice... thanks for making it opt-in with the comment, that's super useful

**Pull request merged**